js/ui/scheduler/workspaces/view_model/utils.js

```diff
@@ -25,3 +25,35 @@ export const getViewDataGeneratorByViewType = (viewType) => {
             return new ViewDataGenerator();
     }
 };
+
```

Comment:
Your solution is nice, 
but I want to suggest a simpler (from my point of view) solution:

```typescript
const DAYS_IN_WEEK = 7;
const MIN_WEEK_IN_MONTH = 6;
const MS_IN_DAY = 24 * 60 * 60 * 1000;

export const resetDateTime = (date) => {
    const dateCopy = new Date(date);
    dateCopy.setHours(0, 0, 0, 0);
    return dateCopy;
};

export const getDaysBetweenDates = (fromDate, toDate) => {
    const msDiff = resetDateTime(toDate).getTime() - resetDateTime(fromDate).getTime();
    // NOTE: We're adding the 1 here because:
    // (fromDate: 2023-01-01, toDate: 2023-01-01) => 1 day
    // (fromDate: 2023-01-01, toDate: 2023-01-03) => 3 days
    return Math.ceil(msDiff / MS_IN_DAY) + 1;
};

export const getDaysCountInWeek = (fromDay, firstDayOfWeek) =>
    fromDay > firstDayOfWeek
        ? 7 - fromDay
        : firstDayOfWeek - fromDay;

export const getWeeksCount = (fromDate, toDate, firstDayOfWeek) => {
    const daysCount = getDaysBetweenDates(fromDate, toDate);
    const daysCountInFirstWeek = getDaysCountInWeek(fromDate.getDay(), firstDayOfWeek);
    const daysCountWithoutFirstWeek = daysCount - daysCountInFirstWeek;

    const weekCount = Math.ceil(daysCountWithoutFirstWeek / DAYS_IN_WEEK) + (daysCount ? 1 : 0);
    return Math.max(weekCount, MIN_WEEK_IN_MONTH);
};
```

I tested it with your jest test cases and looks like everything works as expected :)